compiler/rustc_infer/src/infer/combine.rs

```diff
@@ -43,7 +43,7 @@ use rustc_span::{Span, DUMMY_SP};
 
 #[derive(Clone)]
 pub struct CombineFields<'infcx, 'tcx> {
```

Comment:
can you change all uses of `CombineFields` to `&'a CombineFields<'a, 'tcx>`? don't need 2 separate lifetimes for that

Reply:
I wasn't able to get this to work.

Reply:
ok :thinking: that's surprising :sweat_smile: what are the errors if you do that?

Reply:
I have to use `&'a self` on methods like `fn lub(&'a self, ..) -> Lub<'a, 'tcx>` on `impl CombineFields` and then then after doing that I get `lifetime may not live long enough` where those methods are used.